File: June Challenge/Week 3/Java/Day_18/H-Index-II.java
// Problem : https://leetcode.com/problems/h-index-ii/
// Solution :
class Solution {
    public int hIndex(int[] citations) {
        if(citations.length==0)
            return 0;
        
        int l = 0;
        int r = citations.length-1;
        int n = r + 1 ;
        while(l<=r){
            int mid = (l+r)/2;
            if(citations[mid] >= n-mid ){
                r = mid - 1;
            }
            else{
                l = mid + 1;
            }
        }
        return n - l;
    }
}

File: June Challenge/Week 3/C++/Day_18/HIndexII.cpp
class Solution {
public:
    int hIndex(vector<int>& citations) {
        int N=citations.size();
        if(!N){
            return 0;
        }
        for(int i=0;i<N;i++){
            if(citations.at(i)>=N-i){
                return N-i;
            }
        }
        return 0;
    }
};
